### DISCUSSION TITLE
Twig Component with Stimulus Attributes

### DISCUSSION THREAD

**Discussion by TomoyaHamasaki** [symfony/ux#2915] [Questions & Answers]
How does one use stimulus when using a component?

`<twig:Component {{ stimulus_controller('example') }} />`

I can't do this as I can with a typical element. Is the only way to achieve this goal is by doing this?

`<twig:Component data-controller="example" />`

**Comment by WebMamba** [comment]
~~Yes the HTML syntax allows you to only add attribute to your component and can't parse twig directly like that~~

**Comment by smnandre** [comment]
> `<twig:Component {{ stimulus_controller('example') }} />`

@TomoyaHamasaki  here you can use the `...` syntax 

```twig
<twig:Component {{ ...stimulus_controller('example') }} />
```

:)

**Comment by TomoyaHamasaki** [comment]
+

> > `<twig:Component {{ stimulus_controller('example') }} />`
> 
> @TomoyaHamasaki here you can use the `...` syntax
> 
> ```
> <twig:Component {{ ...stimulus_controller('example') }} />
> ```
> 
> :)

I love you, but why does that work? lol what is `...`?

**Comment by norkunas** [comment]
It's a spread operator (destructuring) borrowed from JavaScript

**Comment by jochienabuurs** [comment]
I came here during a google search. I'm locked into Symfony 5.4 and can't use the latest versions of Twig. I think the spread operator is not supported with the version I'm using. I get the error "("Only arrays and Traversables can be unpacked")" when doing this.
Is there another way or am y doing it wrong?
